vm.tf

```diff
@@ -0,0 +1,35 @@
+resource "azurerm_virtual_machine" "vm" {
+  count                 = var.vm_count
+  name                  = "${var.prefix}-vm-${count.index}"
+  location              = azurerm_resource_group.example.location
+  resource_group_name   = azurerm_resource_group.example.name
+  network_interface_ids = [azurerm_network_interface.main["nic${count.index + 1}"].id]
+  vm_size               = "Standard_DS1_v2"
+
+  storage_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts"
+    version   = "latest"
+  }
+  storage_os_disk {
+    name              = "myosdisk-${count.index}"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Standard_LRS"
+  }
+  os_profile {
+    computer_name  = "hostname-${count.index}"
+    admin_username = "testadmin"
+    admin_password = "Password1234!"
```

Comment:
It's not recommended to hardcode sensitive information like `admin_password` directly in the configuration. Consider using a more secure method, such as retrieving the password from a secure source or using Terraform variables with sensitive attributes.